src/components/ProviderPreview.tsx

```diff
@@ -0,0 +1,80 @@
+'use client'
+
+import Link from 'next/link'
+
+import type { Provider } from '@/payload-types'
+
+import { courseTypesData } from '@/constants/courseTypes'
+import { cn } from '@/utilities/ui'
+import { isValidFullUrl } from '@/utilities/validateUrl'
+import { ExternalLink, Mail, MapPin, Phone } from 'lucide-react'
+import { Dialog, DialogContent, DialogTrigger } from './ui/dialog'
+
+export const ProviderPreview = (props: { doc?: Provider }) => {
+  const { doc } = props
+
+  if (!doc) return null
+
+  const { name, details, location, website, email, phone, courseTypes } = doc
+
+  const getLocationText = () => {
+    if (!location) return null
+    if (location.city && location.state) return `${location.city}, ${location.state}`
+    if (location.city) return location.city
+    if (location.state) return location.state
+    return 'Location'
+  }
+
+  const locationText = getLocationText()
+
+  const courseTypeLabels = courseTypes
+    ?.map((ct) => courseTypesData.find((ctd) => ctd.value === ct)?.label)
+    .filter(Boolean)
+
+  const validWebsite = website && isValidFullUrl(website) ? website : null
+
+  const contactItems = [locationText, validWebsite, email, phone].filter(Boolean)
+
+  return (
+    <Dialog>
+      <DialogTrigger className="cursor-help hover:underline text-base w-fit">{name}</DialogTrigger>
+      <DialogContent className="flex flex-col gap-2 py-2" overlayClassName="bg-transparent">
+        {courseTypeLabels && courseTypeLabels.length > 0 && (
+          <p>Offers: {courseTypeLabels.join(', ')}</p>
```

Comment:
⛏️ If you're in here and want to wrap `Offers:` in a different font weight, that would rock.